File: decompiled/3.18.0/be/ehealth/business/mycarenetcommons/mapper/v3/BlobMapper.java
package be.ehealth.business.mycarenetcommons.mapper.v3;

import be.ehealth.business.mycarenetdomaincommons.domain.Blob;
import be.fgov.ehealth.mycarenet.commons.core.v3.BlobType;

public final class BlobMapper {
   private BlobMapper() {
      throw new UnsupportedOperationException("only static methods may be used for BlobMapper");
   }

   public static BlobType mapBlobTypefromBlob(Blob blob) {
      BlobType result = new BlobType();
      result.setValue(blob.getContent());
      result.setId(blob.getId());
      result.setContentEncoding(blob.getContentEncoding());
      result.setHashValue(blob.getHashValue());
      result.setContentType(blob.getContentType());
      result.setContentEncryption(blob.getContentEncryption());
      return result;
   }

   public static Blob mapBlobfromBlobType(BlobType blob) {
      Blob result = new Blob();
      result.setContent(blob.getValue());
      result.setId(blob.getId());
      result.setContentEncoding(blob.getContentEncoding());
      result.setContentEncryption(blob.getContentEncryption());
      result.setHashValue(blob.getHashValue());
      result.setContentType(blob.getContentType());
      return result;
   }
}

File: src/main/kotlin/org/taktik/connector/business/mycarenetcommons/mapper/v3/BlobMapper.kt
package org.taktik.connector.business.mycarenetcommons.mapper.v3

import org.taktik.connector.business.mycarenetdomaincommons.domain.Blob
import be.fgov.ehealth.mycarenet.commons.core.v3.BlobType

object BlobMapper {
    fun mapBlobTypefromBlob(blob: Blob): BlobType {
       return BlobType().apply {
            value = blob.content
            id = blob.id
            contentEncoding = blob.contentEncoding
            hashValue = blob.hashValue
            contentType = blob.contentType
            contentEncryption = blob.contentEncryption
        }
    }

    fun mapBlobfromBlobType(blob: BlobType): Blob {
        return Blob().apply {
            content = blob.value
            id = blob.id
            contentEncoding = blob.contentEncoding
            contentEncryption = blob.contentEncryption
            hashValue = blob.hashValue
            contentType = blob.contentType
        }
    }
}
